pkg/cmd/get/secrets.go

```diff
@@ -144,69 +151,94 @@ func printPackageSecrets(ctx context.Context, outWriter io.Writer, kubeClient cl
+		if err != nil {
+			return fmt.Errorf("listing secrets: %w", err)
+		}
+
+		for i := range cnoeLabelSecrets.Items {
+			secrets = append(secrets, generateSecret(cnoeLabelSecrets.Items[i], false))
 		}
 
-		for j := range secrets.Items {
-			secretsToPrint = append(secretsToPrint, secretToTemplateData(secrets.Items[j]))
+		if len(secrets) == 0 {
+			fmt.Println("no secrets found")
+			return nil
 		}
 	}
 
-	return printOutput(secretTemplatePath, outWriter, secretsToPrint, format)
+	return printSecretsOutput(outWriter, secrets, format)
 }
 
-func renderTemplate(templatePath string, outWriter io.Writer, data []any) error {
-	tmpl, err := templates.ReadFile(templatePath)
-	if err != nil {
-		return fmt.Errorf("failed to read template: %w", err)
+func generateSecretTable(secretTable []Secret) metav1.Table {
+	table := &metav1.Table{}
+	table.ColumnDefinitions = []metav1.TableColumnDefinition{
+		{Name: "Name", Type: "string"},
+		{Name: "Namespace", Type: "string"},
+		{Name: "Username", Type: "string"},
+		{Name: "Password", Type: "string"},
+		{Name: "Token", Type: "string"},
+		{Name: "Data", Type: "string"},
 	}
+	for _, secret := range secretTable {
+		var dataEntries []string
 
-	t, err := template.New("secrets").Parse(string(tmpl))
-	if err != nil {
-		return fmt.Errorf("parsing template: %w", err)
-	}
-	for i := range data {
-		tErr := t.Execute(outWriter, data[i])
-		if tErr != nil {
-			return fmt.Errorf("executing template for data %s : %w", data[i], tErr)
+		if !secret.isCore {
+			for key, value := range secret.Data {
+				dataEntries = append(dataEntries, fmt.Sprintf("%s=%s", key, value))
+			}
+		}
+		dataString := strings.Join(dataEntries, ", ")
+		row := metav1.TableRow{
+			Cells: []interface{}{
+				secret.Name,
+				secret.Namespace,
+				secret.Username,
+				secret.Password,
+				secret.Token,
+				dataString,
+			},
 		}
+		table.Rows = append(table.Rows, row)
 	}
-	return nil
+	return *table
 }
 
-func printOutput(templatePath string, outWriter io.Writer, data []any, format string) error {
+func printSecretsOutput(outWriter io.Writer, secrets []Secret, format string) error {
 	switch format {
 	case "json":
-		enc := json.NewEncoder(outWriter)
-		enc.SetEscapeHTML(false)
-		enc.SetIndent("", "  ")
-		return enc.Encode(data)
+		return util.PrintDataAsJson(secrets, outWriter)
 	case "yaml":
-		b, err := yaml.Marshal(data)
-		if err != nil {
-			return err
-		}
-		_, err = outWriter.Write(b)
-		return err
-	case "":
-		return renderTemplate(templatePath, outWriter, data)
+		return util.PrintDataAsYaml(secrets, outWriter)
+	case "", "table":
+		return util.PrintTable(generateSecretTable(secrets), outWriter)
 	default:
 		return fmt.Errorf("output format %s is not supported", format)
 	}
 }
 
-func secretToTemplateData(s v1.Secret) TemplateData {
-	data := TemplateData{
+func generateSecret(s v1.Secret, isCoreSecret bool) Secret {
+	secret := Secret{
 		Name:      s.Name,
 		Namespace: s.Namespace,
```

Comment:
We should name this `populateSecrets` instead of `generateSecret`. I felt we are creating a new secret.

Reply:
Fixed: 813c0cd